Datasets are huge so previously created a subset of each.

zcat B3.error_detected.dge.txt.gz | awk -F"\t" '{for(i=1;i<=100;i++)printf("%s\t",$i);printf("\n")}' | head -100 | gzip > B3.100.error_detected.dge.txt.gz
zcat B4.error_detected.dge.txt.gz | awk -F"\t" '{for(i=1;i<=100;i++)printf("%s\t",$i);printf("\n")}' | head -100 | gzip > B4.100.error_detected.dge.txt.gz

Read them each in.
Them merge them.

In [1]:
getwd()

[1] "/Users/jakewendt/github/unreno/syryu/notebooks"

In [2]:
list.files( pattern = "*error_detected.dge.txt.gz" )

[1] "B3.100.error_detected.dge.txt.gz"   "B3.1000.error_detected.dge.txt.gz" 
[3] "B3.10000.error_detected.dge.txt.gz" "B3.error_detected.dge.txt.gz"      
[5] "B4.100.error_detected.dge.txt.gz"   "B4.10000.error_detected.dge.txt.gz"
[7] "B4.error_detected.dge.txt.gz"

In [3]:
library(Seurat)

Loading required package: ggplot2
Loading required package: cowplot

Attaching package: ‘cowplot’

The following object is masked from ‘package:ggplot2’:

    ggsave

Loading required package: Matrix


Files 100 and 1000 didn't work when trying to create seurat object?

In [4]:
B3.data <- read.table("B3.10000.error_detected.dge.txt.gz",row.names=1,header=T)

In [5]:
nrow(B3.data)

[1] 9999

In [6]:
ncol(B3.data)

[1] 9999

In [7]:
B3 <- CreateSeuratObject(raw.data = B3.data, min.cells = 3,  min.genes = 200, project = 'B3')

In [8]:
B4.data <- read.table("B4.10000.error_detected.dge.txt.gz",row.names=1,header=T)

In [9]:
B4 <- CreateSeuratObject(raw.data = B4.data, min.cells = 3,  min.genes = 200, project = 'B4')

In [10]:
merged <- MergeSeurat(object1 = B3, object2 = B4, add.cell.id1 = "B3", add.cell.id2 = "B4", project = "B3-B4")

In [11]:
head(x = merged@cell.names)

[1] "B3_CGCTCGGCGACT" "B3_TGACTCAGCTGG" "B3_CATGTCAGCAAT" "B3_TAACCTCGGGCC"
[5] "B3_CTGGCCATTCGG" "B3_TCTGCCGCATCG"

In [12]:
tail(x = merged@cell.names)

[1] "B4_GTGGCGGTCAGT" "B4_TCGCGGTGTGGA" "B4_GGCTCTCAGCGG" "B4_GATGCGTTTACA"
[5] "B4_GTCAGACTAGGT" "B4_GCGCTGTCAGGA"

In [13]:
table(merged@meta.data$orig.ident)


 B3  B4 
581 862 

AddSamples seems to require the same number of rows.
This error message seems to be wrong though as my samples have only 9999 rows and 9999 columns?

Error in data.frame(..., check.names = FALSE): arguments imply differing number of rows: 10562, 10580

In [15]:
# added_to <- AddSamples(object = B3, new.data = B4.data, add.cell.id = "B4")

In [22]:
head(x = merged@data)

   [[ suppressing 1443 column names ‘B3_CGCTCGGCGACT’, ‘B3_TGACTCAGCTGG’, ‘B3_CATGTCAGCAAT’ ... ]]


6 x 1443 sparse Matrix of class "dgCMatrix"
                                                                            
0610007P14Rik .        1.950898 .        .        .        .        .       
0610009B22Rik 1.687737 1.950898 2.074296 2.105329 2.117453 2.142275 2.242571
0610009D07Rik 2.283856 1.950898 .        .        .        .        2.242571
0610009E02Rik .        .        .        .        .        .        .       
0610009L18Rik .        .        .        .        .        .        .       
0610009O20Rik .        .        .        .        2.117453 .        .       
                                                                            
0610007P14Rik .       . . 2.312454 . .        . 2.506445 .        2.388854 .
0610009B22Rik .       . . .        . .        . 3.157958 2.455274 2.388854 .
0610009D07Rik 2.28109 . . 2.312454 . 2.427505 . 2.506445 .        2.388854 .
0610009E02Rik .       . . .        . .        . .        .        .        .
0610009L18Rik .       . . .     

In [24]:
merged@cell.names

[1] "B3_CGCTCGGCGACT" "B3_TGACTCAGCTGG" "B3_CATGTCAGCAAT" "B3_TAACCTCGGGCC"
   [5] "B3_CTGGCCATTCGG" "B3_TCTGCCGCATCG" "B3_AGCGAGTAGCCT" "B3_TACCGGATGAAG"
   [9] "B3_GAGCATAATAAT" "B3_GTCCTGGATTCT" "B3_TGCAGCCCAGGT" "B3_TTCCCAGTCGAA"
  [13] "B3_AGAGGTGTGTAA" "B3_ACTCGTAGGCTC" "B3_TTAGTCATGTTT" "B3_AGCTTCTCACTG"
  [17] "B3_GCTCCTCCAGTC" "B3_GTCTAGTTATCC" "B3_CCATCAGTAACG" "B3_TACGGCACTAGC"
  [21] "B3_AGGTCGTTCGTC" "B3_TCCACGGATCCC" "B3_GTGTTGTTAGGA" "B3_TCACGATTCGAG"
  [25] "B3_GATCCTTGCGGC" "B3_GAGGCGTACGTT" "B3_TACTCTCGAGTC" "B3_GATCAACCTGAG"
  [29] "B3_GGATATTTGGTC" "B3_GCTCCTGATCTT" "B3_ACGTTAAGTCGT" "B3_TTGAATACGCTA"
  [33] "B3_AGTTGCTTCTGC" "B3_AAGTAAATAGCC" "B3_CTGCTGCCCAGT" "B3_ACGACTGCGGCA"
  [37] "B3_CACACGTATTAA" "B3_TGCAAGTTCGAG" "B3_CAGTATAGCGTG" "B3_CTTTCTCAGTCT"
  [41] "B3_CTCACATTAACA" "B3_CAGGCTCCTTCG" "B3_GCCTCGGCCGCC" "B3_CTCTTCAAGCAC"
  [45] "B3_TCCTTCAGTCTC" "B3_GGCTCCCCATAC" "B3_ATCTCATGACAT" "B3_GTTTGCCAGTTT"
  [49] "B3_ATTTAGGCTTTC" "B3_TAATGATGTTGC" "B3_TGACTTTATCCG" "B3_CAGAAAGATGGC"
  [53] "B3_TGGACCTTTACG" "B3_CACAACAATGTG" "B3_CACGGAGTAGTA" "B3_GAGCAGCCCTCA"
  [57] "B3_TACGTGTTCTAG" "B3_CTGGTCATAGGG" "B3_GTCGGGATGTTG" "B3_TATGGCTTACAC"
  [61] "B3_TGCCCGCTCGTC" "B3_TTACGAGGAAAG" "B3_ATATACTAAGCT" "B3_CAGGTAGATCGC"
  [65] "B3_CATCATCGAGCC" "B3_GTATGACTTCTG" "B3_ATCCTTAGCCCT" "B3_ATGGCCTGCCCT"
  [69] "B3_AGTTTAGAGTCT" "B3_TGAGGGTATGAC" "B3_GACGCCTCACAT" "B3_GAGACGCCAGGG"
  [73] "B3_GACGTAGATCCA" "B3_AGCCGACCGCGA" "B3_TGTCAACTTTTT" "B3_AGAGGTCGCCAA"
  [77] "B3_TTCCTAAGCGGA" "B3_GTTTATCTTCGT" "B3_TGATGACATATT" "B3_AAACCGGCAACT"
  [81] "B3_ATACGACTGTTG" "B3_TCCCAGATTATT" "B3_AATGGCCGATTT" "B3_CCCGTCACTTCC"
  [85] "B3_GAGAAGAGTTTT" "B3_GGCATGTCCGCC" "B3_CGAAGCGAGGGC" "B3_TTGTATCATGGA"
  [89] "B3_CAGCTAGCAGCT" "B3_ATCGTCCCATAT" "B3_AACTGCGCCTTC" "B3_TACTAGAACCCA"
  [93] "B3_CTTGATCGCGCA" "B3_TAAGAATCTAGG" "B3_ACATTAGAGTGC" "B3_ACTCATAGCCCT"
  [97] "B3_ACTCGCCAGGTA" "B3_GGAACACATTCG" "B3_TGGGAATGGCAT" "B3_GGACATTTATCA"
 [101] "B3_GAGATGAATTCC" "B3_TCTTTCTCAGGC" "B3_TAGTCAAATGTG" "B3_TTAGTACATCAT"
 [105] "B3_TTTATATCTCTT" "B3_TACGTGTTATAG" "B3_GAAAGTGCTGTC" "B3_CTATGGCTAAAT"
 [109] "B3_TGACGTGCTGGC" "B3_AGTTATAACTTT" "B3_CAGTATACGTGG" "B3_ACGGCGCGTGTG"
 [113] "B3_AAGATCCATCAG" "B3_TGTCGTCACGCG" "B3_ACCACATTATCC" "B3_ACAAAATACCTT"
 [117] "B3_GCGATTTATGAG" "B3_TTATTGTTCCAT" "B3_CCGGTAGAAAAT" "B3_GGGCTCCGAGTT"
 [121] "B3_TCCTCCAGTGAA" "B3_CAACGTCGAATC" "B3_CTCAACTTGAGA" "B3_TACTGTTACTGA"
 [125] "B3_CCTAGTTGTACT" "B3_GGCTTTCGGGTC" "B3_TGCTAGCTACAG" "B3_GCCCAAGTGGCT"
 [129] "B3_TCGCAATTGCGG" "B3_CCATTTCGCCCG" "B3_CCATCTGAAGCA" "B3_GCCCAAAGTTTT"
 [133] "B3_CTGTACTGCCTC" "B3_CGACCTCGGCAC" "B3_GCGTGACGCTTC" "B3_GTTCATATGGGA"
 [137] "B3_TTGTCATGATAA" "B3_ACACATTTTTTC" "B3_GAGGGTCCGACG" "B3_TGCCCCATTCGA"
 [141] "B3_CTACCTTACTCG" "B3_CAGTTATAGTCC" "B3_CACGATAAACCA" "B3_GACGGTAGCAAG"
 [145] "B3_CCTCAGAACTAG" "B3_TCGCCTGTCCGC" "B3_GGGTAAGGTCAG" "B3_CCGATAACCGCC"
 [149] "B3_CTCGTGTCGAGA" "B3_TTCAGGGTAGGT" "B3_TAAACGGCCTGT" "B3_TCGTCCCTTAAG"
 [153] "B3_ATGTAATCGAGT" "B3_AATGCATGACGA" "B3_CATTAATCGCGT" "B3_CAAAGACTTGTT"
 [157] "B3_CCCCATCGCTTA" "B3_AAACGATCCTTC" "B3_TGTAGATATGAC" "B3_TCTATGGCAAAT"
 [161] "B3_CAGGTCGGACAA" "B3_GAATTGTTTGCC" "B3_GCGCCCCTCTCC" "B3_GGAAGGTGCTGT"
 [165] "B3_GCCCCACTCTTT" "B3_CCCCCAAAATGT" "B3_CATCATTCAGTC" "B3_TTAATCAAGCAG"
 [169] "B3_GCTCGCGTCCAT" "B3_CATGTGTGAGGT" "B3_AAGGTAGCGGCG" "B3_TGACGGCTAGGT"
 [173] "B3_CATGTCGATCAT" "B3_GGCACGGGCGTG" "B3_CGGTCCCCTTCA" "B3_CTCAGTGCTATT"
 [177] "B3_AAGACGACACCG" "B3_CTATATCGTGAC" "B3_CTCATGCAGTTT" "B3_GCGTACCCGCTT"
 [181] "B3_GGCGACTGGACT" "B3_ACCTCTCATTTT" "B3_AGTTCCGACTAT" "B3_TGAACTCCTAAG"
 [185] "B3_CATACTCATAAC" "B3_CCAAGCTTACTA" "B3_GCATGGAAACTC" "B3_AGGTTGGCATGC"
 [189] "B3_GCGCCCACCTCA" "B3_ACCTGAGAGCGC" "B3_CGACGCCCTCGG" "B3_CTCCCTGACATC"
 [193] "B3_TAACCTATCAGT" "B3_GCAAAGGGCACC" "B3_CAATCTTTGAAG" "B3_ACACCCGCGTTC"
 [197] "B3_CCGTATCAACTA" "B3_GCAAAGGCCGGC" "B3_GGAACAGTTAAT" "B3_TCAAATCACCCC"
 [201] "B3_TGTTCAGAGACC" "B3_TCCGCTCCAAAC" "B3_CCCTGC